From: Yevgeny Petrilin <yevgenyp@mellanox.co.il>
Date: Mon, 19 Dec 2011 04:03:05 +0000
Subject: mlx4: not using spin_lock_irq when getting vf by resource.
Patch-mainline: v3.3-rc1
Git-commit: 996b0541e73a3321947dbc8894a078b8e82e8691
References: bnc#786036 FATE#314304

The function is always called from irq context, changing the call
to spin_lock().

Signed-off-by: Yevgeny Petrilin <yevgenyp@mellanox.co.il>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/mellanox/mlx4/resource_tracker.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
--- a/drivers/net/ethernet/mellanox/mlx4/resource_tracker.c
+++ b/drivers/net/ethernet/mellanox/mlx4/resource_tracker.c
@@ -339,14 +339,14 @@ int mlx4_get_slave_from_resource_id(stru
 
 	if (type == RES_QP)
 		id &= 0x7fffff;
-	spin_lock_irq(mlx4_tlock(dev));
+	spin_lock(mlx4_tlock(dev));
 
 	r = find_res(dev, id, type);
 	if (r) {
 		*slave = r->owner;
 		err = 0;
 	}
-	spin_unlock_irq(mlx4_tlock(dev));
+	spin_unlock(mlx4_tlock(dev));
 
 	return err;
 }
